### PR TITLE
refactor: create SuperDCAPoolStorage contract and separate storage tests

### PR DESCRIPTION
**Pull request by ely67**
## Summary
This PR introduces a new `SuperDCAPoolStorage` contract to separate storage concerns from the main pool logic and adds comprehensive test coverage for storage functionality.

### 🏗️ **Contract Architecture**
- **Refactored `SuperDCAPoolV1.sol`** - Now inherits from `SuperDCAPoolStorage` for cleaner separation of concerns
- **Moved view functions** from `SuperDCAPoolV1` to `SuperDCAPoolStorage` including:
  - `getIDAShares()`, `getIDAIndexValue()`
  - `getTradeInfo()`, `getLatestTrade()`, `getTradeCount()`
  - `getNextDistributionTime()`, `getExecutionFeeShare()`
  - Helper functions: `_isInputToken()`, `_isOutputToken()`, `_getUnderlyingToken()`

### 🧪 **Test Suite Improvements**
- **Added `test/SuperDCAPoolStorage.t.sol`** - Comprehensive test suite with 18+ test functions covering:
  - Agreement type validation (`_isCFAv1`, `_isIDAv1`)
  - Token validation and underlying token mapping
  - Price feed functionality and edge cases
  - Fee calculation and user uninvested calculations
  - Constants, state variables, and error handling
  - Next distribution time calculations with various scenarios

- **Refactored `test/SuperDCAPoolV1.t.sol`** - Removed tests that now belong in storage tests to avoid duplication

### 🎯 **Benefits**
- **Better separation of concerns** - Storage logic isolated from business logic
- **Improved testability** - Storage functions can be tested in isolation
- **Cleaner architecture** - Easier to maintain and extend
- **Reduced code duplication** - Clear boundaries between test responsibilities

## Testing
- ✅ All existing tests pass
- ✅ No breaking changes to existing functionality

### CODE REVIEW

File: test/SuperDCAPoolV1.t.sol [hunk 1]
```diff
@@ -457,55 +458,14 @@ contract SuperDCAPoolV1Test is Test {
     _createFlow(alice, USDCX, address(pool), uint96(INFLOW_RATE_USDC));
 
     // Should revert when trying to close stream with sufficient balance
-    vm.expectRevert(SuperDCAPoolV1.NotClosable.selector);
+    vm.expectRevert(SuperDCAPoolStorage.NotClosable.selector);
     pool.closeStream(alice, ISuperToken(USDCX));
   }
 
-  function testFork_ExecutionFeeShareAdjustment() public {
```
Comment (mikeghen) [inline test/SuperDCAPoolV1.t.sol:464]: This test should be kept. I didn't see it tested elsewhere.

**Review by mikeghen — CHANGES_REQUESTED** [review]
This looks good to me @ely67, nice work 👏 

I have one request to bring back a test I think you dropped out of the suite at some point. Let me know if I missed a reason why we would remove that test.

Also look at the merge conflicts.

**Comment by mikeghen** [comment]
@ely67, thanks for the contribution but this has gotten stale so I'm closing it.

**Comment by ely67** [comment]
ok

**Comment by ely67** [comment]
@mikeghen What’s the best way to get in touch with you outside of the github?email, linkedin or something else?
